content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,300 @@
+Path to a metrics file [metrics.json, n to omit]:
+Path to a plots file/directory [plots, n to omit]: logs.csv
+──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+default:
+  cmd: python src/train.py
+  deps:
+  - data/images/
+  - src/train.py
+  params:
+  - model
+  - train
+  outs:
+  - models
+  metrics:
+  - metrics.json:
+      cache: false
+  plots:
+  - logs.csv:
+      cache: false
+Do you want to add the above contents to dvc.yaml? [y/n]: y
+
+Created default stage in dvc.yaml. To run, use "dvc exp run".
+See https://dvc.org/doc/user-guide/experiment-management/running-experiments.
+```
+
+DVC treats experiments as code. Rather than log parameters and metrics with API
+calls, DVC expects them to be saved in files. This might mean a little more work
+to set up your code to read parameters from a YAML file or write metrics to a
+JSON file. However, once you set up your repo in this structure, you start to
+see the benefits of this approach.
+
+# Benefits of Versioning
+
+With an experiment tracking API, you insert API calls in your code to save
+experiment information in a central database. With experiment versioning,
+experiment information lives in readable files that are always available, and
+your code stays clean.
+
+Even without DVC, you can read, save, and version your experiment parameters and
+metrics:
+
+```dvc
+$ cat params.yaml
+train:
+  epochs: 10
+model:
+  conv_units: 128
+```
+
+```dvc
+$ cat metrics.json
+{"loss": 0.24310708045959473, "acc": 0.9182999730110168}
+```
+
+You can see what changed in parameters, code, or anything else:
+
+```dvc
+$ git diff HEAD~1 -- params.yaml
+diff --git a/params.yaml b/params.yaml
+index baad571a2..57d098495 100644
+--- a/params.yaml
++++ b/params.yaml
+@@ -1,5 +1,5 @@
+ train:
+   epochs: 10
+-model:
+-  conv_units: 16
++model:
++  conv_units: 128
+```
+
+Many experiment tracking databases do not capture everything you need to
+reproduce an experiment, especially the pipeline to run the experiment end to
+end. With experiment versioning, it's important to codify your experiment
+pipeline so that anyone can reproduce the experiment. Notice that the
+`dvc exp init` command above generated a `dvc.yaml` file that describes the
+experiment pipeline:
+
+```dvc
+$ cat dvc.yaml
+stages:
+  default:
+    cmd: python src/train.py
+    deps:
+    - data/images
+    - src/train.py
+    params:
+    - model
+    - train
+    outs:
+    - models
+    metrics:
+    - metrics.json:
+        cache: false
+    plots:
+    - logs.csv:
+        cache: false
+```
+
+This captures:
+
+- The command to run the experiment.
+- Parameters and other dependencies.
+- Metrics, plots, and other outputs.
+
+The pipeline has a single `default` stage, but more stages can be added.
+
+With an experiment tracking dashboard, you see all your experiments, and you may
```

Comment:
Will it make sense without mentioning `dvc exp show` or DVC specifically? Do we show it again when mentioning how DVC helps? Maybe it's better to remove this sentence and put it in the next section?

Reply:
> without mentioning dvc exp show or DVC specifically? 

It might be. You can mention that this is an idea. Or say that you will show later how to create it.